dask/bytes/core.py

```diff
@@ -112,13 +112,31 @@ def read_bytes(
                     "Backing filesystem couldn't determine file size, cannot "
                     "do chunked reads. To read, set blocksize=None."
                 )
-            off = list(range(0, size, blocksize))
-            length = [blocksize] * len(off)
-            if not_zero:
-                off[0] = 1
-                length[0] -= 1
-            offsets.append(off)
-            lengths.append(length)
+
+            elif size == 0:
+                # skip empty
+                offsets.append([])
+                lengths.append([])
+            else:
+                # shrink blocksize to give same number of parts
+                if size % blocksize and size > blocksize:
+                    blocksize = size / (size // blocksize)
```

Comment:
Please check: I think if I rewrite `blocksize` here, I risk reducing it on every file iteration, right? I should have a new local variable here, right?

Reply:
Oh, yes! Sorry I didn't catch that, you're right.